herzog/__init__.py

```diff
@@ -38,10 +38,53 @@ def __exit__(self, *args, **kwargs):
                     del __main__.__dict__[key]
             __main__.__dict__.update(self._state_modules)
 
-def generate(handle: TextIO):
-    cells = [obj.to_ipynb_cell() for obj in parse_cells(handle)
+def load_ipynb_cells(ipynb: TextIO) -> List[Dict[Any, Any]]:
+    try:
+        return json.loads(ipynb.read())['cells']
+    except (json.JSONDecodeError, KeyError):
+        print(f"Check that '{ipynb}' is a valid ipynb file.", file=sys.stderr)
+        raise
+
+def translate_to_ipynb(herzog_handle: TextIO) -> Dict[str, Any]:
+    cells = [obj.to_ipynb_cell() for obj in parse_cells(herzog_handle)
              if obj.has_ipynb_representation]
     with open(os.path.join(os.path.dirname(__file__), "data", "python_3_boiler.json")) as fh:
         boiler = json.loads(fh.read())
-    ipynb = dict(cells=cells, **boiler)
-    return ipynb
+    return dict(cells=cells, **boiler)
+
+def translate_to_herzog(ipynb_handle: TextIO, indent: int = 4) -> Generator[str, None, None]:
+    cells = load_ipynb_cells(ipynb_handle)
+    prefix = " " * indent
+    yield "import herzog\n\n"
+
+    for cell in cells:
+        if isinstance(cell.get('source', None), list):
+            cell['source'] = "".join(cell['source'])
+
+        if cell['cell_type'] == "markdown":
+            s = '\nwith herzog.Cell("markdown"):\n    """\n'
+            s += textwrap.indent(cell['source'], prefix=prefix).rstrip()
+            s += '\n    """\n'
+            for line in s.split("\n"):
+                yield line + "\n"
+        elif cell['cell_type'] == "code":
+            s = "\nwith herzog.Cell('python'):\n"
+            s += textwrap.indent(cell['source'], prefix=prefix).rstrip()
+            for line in s.split("\n"):
+                if line.startswith("%"):
+                    yield line.replace("%", JUPYTER_MAGIC_PFX, 1) + "\n"
+                elif line.startswith("#"):
+                    yield line.replace("#", JUPYTER_SHELL_PFX, 1) + "\n"
```

Comment:
The Jupyter shell prefix character is !

e.g.
```
!gsutil cp foo gs://bar
```

Reply:
Oof, my bad.  Thanks for the catch!